lib/src/facts/resolvers/operating_system_resolver.cc

```diff
@@ -57,6 +57,18 @@ namespace facter { namespace facts { namespace resolvers {
         if (!data.release.empty()) {
             auto value = make_value<map_value>();
 
+            // When we have no major or minor, do a 'trivial' parse of
+            // the release to try to get SOMETHING out of it.
+            if(data.minor.empty() && data.major.empty()) {
+                 auto begin = data.release.begin();
+                 auto end = data.release.end();
+                 auto end_first = find(begin, end, '.');
+                 auto begin_second = next(end_first);
+                 auto end_second = find(begin_second, end, '.');
+                 data.major = std::string(begin, end_first);
+                 data.minor = std::string(begin_second, end_second);
```

Comment:
@branan check out https://github.com/whopper/facter/blob/FACT-1056/osx_release/lib/src/facts/osx/operating_system_resolver.cc#L42-50

We've actually had this functionality before one of the OS refactors, and that was how it was originally implemented. Could be a bit cleaner.